src/consensus/pbft_manager.cpp

```diff
@@ -1855,15 +1843,18 @@ std::optional<SyncBlock> PbftManager::processSyncBlock() {
   return std::optional<SyncBlock>(std::move(sync_block.first));
 }
 
-void PbftManager::syncBlockQueuePush(SyncBlock const &block, dev::p2p::NodeID const &node_id) {
+void PbftManager::syncBlockQueuePush(SyncBlock &&block, dev::p2p::NodeID const &node_id) {
+  const auto period = block.pbft_blk->getPeriod();
+  if (period <= sync_period_) return;
```

Comment:
- We should log error msg here
- I think the condition should not check <= but period should be excatly +1 ->  if (period != sync_period_ + 1)